src/groupNormL2.jl

```diff
@@ -0,0 +1,88 @@
+# Group L2 norm (times a constant)
+
+export GroupNormL2
+
+"""
+**``L_2`` Group - norm**
+    GroupNormL2(λ = 1, g = 1, idx = [:])
+Returns the function
+```math
+f(x) =  \\sum\\_{i} \\lambda\\_{i}||x\\_{[i]}||\\_2)^{1/2}
+```
+for groups `x\\_{[i]}` and nonnegative weights `λ\\_i`.
+  This operator reduces to the two norm if only one group is defined.
+"""
+struct GroupNormL2{
+  R <: Real,
+  RR <: AbstractVector{R},
+  I
+  } <: ProximableFunction
+  lambda::RR
+  idx::I
+
+  function GroupNormL2{R, RR,I}(
+    lambda::RR,
+    idx::I
+    ) where {R <: Real, RR <: AbstractVector{R}, I}
+    if any(lambda .< 0)
+      error("weights λ must be nonnegative")
+    elseif length(lambda) != length(idx)
+      error("number of weights and groups must be the same")
+    else
+      new{R, RR, I}(lambda, idx)
+    end
+  end
+end
+
+GroupNormL2(
+  lambda::RR = [1.],
+  idx::I = [:],
+) where {R <: Real, RR <: AbstractVector{R}, I} =
+GroupNormL2{R, RR, I}(lambda, idx)
+
+function (f::GroupNormL2)(x::AbstractArray{R}) where {R <: Real}
+  sum_c = R(0)
+  if length(f.idx) == 1
+    return f.lambda * norm(x)
+  else
+    for i = 1:length(f.idx)
+      sum_c += f.lambda[i]*norm(x[f.idx[i]])
+    end
+    return sum_c
+  end
+end
+
+function prox!(
+  y::AbstractArray{R},
+  f::GroupNormL2{R, RR, I},
+  x::AbstractArray{R},
+  γ::R = R(1),
+) where {R <: Real, RR <: AbstractVector{R}, I}
+
+  ysum = R(0)
+  if length(f.idx) == 1
```

Comment:
This is no longer correct, right? I think you can forget about this first case now, can't you?

Reply:
Yes it should work now I believe.

Reply:
Still don't think we need this case where length is 1.